ARTEMIS-5279 Add remote federation views to management services

tabish121 opened the pull request:
Provide remote broker connections management beans that mirror the management objects added on the local broker when AMQP federation is configured and the connection is active.

gemmellr commented on artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/federation/AMQPFederationTarget.java:
```diff
@@ -137,51 +139,71 @@ protected void signalError(Exception cause) {
    @Override
    void unregisterFederationProdcerManagement(AMQPFederationSenderController sender) throws Exception {
-      // Not yet implemented for the target side of the federation connection
+      if (brokerConnection.isManagable()) {
```
Not the change itself, but the method its in: typo Prodcer -> Producer.

gemmellr commented on artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/federation/AMQPFederationManagementSupport.java:
```diff
@@ -141,10 +172,58 @@ public static ObjectName getFederationSourceObjectName(ManagementService managem
+   }
+
+   public static ObjectName getFederationTargetObjectName(ManagementService management, String remoteNodeId, String brokerConnection, String federationName) throws Exception {
+      final String brokerConnectionName = management.getObjectNameBuilder().getRemoteBrokerConnectionBaseObjectNameString(remoteNodeId, brokerConnection);
```
Bit of a nitpick, but 'brokerConnectionName' keeps making me think its just the connection name, when its simply not...maybe make it 'brokerConnectionObjectName' for clarity ?

(same in the several other places getRemoteBrokerConnectionBaseObjectNameString is called)

gemmellr commented on tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPFederationManagementTest.java:
```diff
@@ -1695,4 +1710,305 @@ public void testRemoteQueueFederationTrackingCleanedUpOnBrokerConnectionStopped(
+
+         peer.waitForScriptToComplete(5, TimeUnit.SECONDS);
+
+         final AMQPFederationRemotePolicyControlType remotePolicyControl = (AMQPFederationRemotePolicyControlType)
+            server.getManagementService().getResource(policyResourceName);
+         AMQPFederationProducerControl producerControl = (AMQPFederationProducerControl)
```
make both final?

gemmellr commented on tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPFederationManagementTest.java:
```diff
@@ -1695,4 +1710,305 @@ public void testRemoteQueueFederationTrackingCleanedUpOnBrokerConnectionStopped(
+
+         peer.waitForScriptToComplete(5, TimeUnit.SECONDS);
+
+         final AMQPFederationRemotePolicyControlType remotePolicyControl = (AMQPFederationRemotePolicyControlType)
+            server.getManagementService().getResource(policyResourceName);
+         AMQPFederationProducerControl producerControl = (AMQPFederationProducerControl)
+            server.getManagementService().getResource(producerResourceName);
```
make both final?

gemmellr commented on tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPFederationManagementTest.java:
```diff
@@ -1695,4 +1710,305 @@ public void testRemoteQueueFederationTrackingCleanedUpOnBrokerConnectionStopped(
+
+         peer.waitForScriptToComplete(5, TimeUnit.SECONDS);
+
+         final AMQPFederationRemotePolicyControlType remotePolicyControl = (AMQPFederationRemotePolicyControlType)
+            server.getManagementService().getResource(policyResourceName);
+         AMQPFederationProducerControl producerControl = (AMQPFederationProducerControl)
+            server.getManagementService().getResource(producerResourceName);
```
make both final?

gemmellr commented on tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPFederationManagementTest.java:
```diff
@@ -1695,4 +1710,305 @@ public void testRemoteQueueFederationTrackingCleanedUpOnBrokerConnectionStopped(
+
+         peer.waitForScriptToComplete(5, TimeUnit.SECONDS);
+
+         final AMQPFederationRemotePolicyControlType remotePolicyControl = (AMQPFederationRemotePolicyControlType)
+            server.getManagementService().getResource(policyResourceName);
+         AMQPFederationProducerControl producerControl = (AMQPFederationProducerControl)
+            server.getManagementService().getResource(producerResourceName);
```
make both final?

tabish121 commented:
I've addressed the comments and rebased on main which required fixing a couple conflicts and fixed a couple warnings leftover from those changes.  Ran the updated branch through CI and all tests passed.
